Invalid stdin error message

Using this file:

~~~go
package m

import (
   "strings"
   "testing"
)

func TestPage(t *testing.T) {
   s := strings.Repeat("Moar\n", 99)
   r := NewReaderFromText("Moar", s)
   e := NewPager(r).Page()
   if e != nil {
      t.Error(e)
   }
}
~~~

I get this result:

~~~
--- FAIL: TestPage (0.00s)
    page_test.go:13: Getting piped data on stdin is not supported on Windows,
fixes needed in here:
https://github.com/walles/moar/blob/walles/fix-windows/twin/screen-setup.go
~~~

However it seems the error message is invalid, as I am feeding data from a string, not from standard input.

--- a/twin/screen-setup-windows.go
+++ b/twin/screen-setup-windows.go
@@ -3,8 +3,8 @@
 package twin
 
 import (
-	"fmt"
 	"os"
+	"syscall"
 
 	"golang.org/x/sys/windows"
 	"golang.org/x/term"
@@ -19,20 +19,14 @@ func (screen *UnixScreen) setupSigwinchNotification() {
 }
 
 func (screen *UnixScreen) setupTtyInTtyOut() error {
-	if !term.IsTerminal(int(os.Stdin.Fd())) {
-		// See the counterpart method in screen-setup.go for inspiration.
-		//
-		// A fix might be centered about opening "CONIN$" as screen.ttyIn rather
-		// than using os.Stdin, but I never got that working fully.
-		// Contributions welcome.
-		return fmt.Errorf("Stdin must be a terminal for now. To fix, go here: https://github.com/walles/moar/blob/master/twin/screen-setup-windows.go")
+	in, err := syscall.Open("CONIN$", syscall.O_RDWR, 0)
+	if err != nil {
+		return err
 	}
 
-	// This won't work if we're getting data piped to us, contributions welcome.
-	screen.ttyIn = os.Stdin
+	screen.ttyIn = os.NewFile(uintptr(in), "/dev/tty")
 
 	// Set input stream to raw mode
-	var err error
 	stdin := windows.Handle(screen.ttyIn.Fd())
 	err = windows.GetConsoleMode(stdin, &screen.oldTtyInMode)
 	if err != nil {

